apidef/oas/authentication.go

```diff
@@ -336,14 +336,32 @@ func (as *AuthSource) ExtractTo(enabled *bool, name *string) {
 
 // Signature holds the configuration for signature validation.
 type Signature struct {
-	Enabled          bool       `bson:"enabled" json:"enabled"` // required
-	Algorithm        string     `bson:"algorithm,omitempty" json:"algorithm,omitempty"`
-	Header           string     `bson:"header,omitempty" json:"header,omitempty"`
-	Query            AuthSource `bson:"query,omitempty" json:"query,omitempty"`
-	Secret           string     `bson:"secret,omitempty" json:"secret,omitempty"`
-	AllowedClockSkew int64      `bson:"allowedClockSkew,omitempty" json:"allowedClockSkew,omitempty"`
-	ErrorCode        int        `bson:"errorCode,omitempty" json:"errorCode,omitempty"`
-	ErrorMessage     string     `bson:"errorMessage,omitempty" json:"errorMessage,omitempty"`
+	// Enabled activates signature validation.
+	// Tyk classic API definition: `auth_configs[X].validate_signature`.
+	Enabled bool `bson:"enabled" json:"enabled"` // required
+	// Algorithm is the signature method to use.
+	// Tyk classic API definition: `auth_configs[X].signature.algorithm`.
+	Algorithm string `bson:"algorithm,omitempty" json:"algorithm,omitempty"`
+	// Header is the name of the header to consume.
+	// Tyk classic API definition: `auth_configs[X].signature.header`.
```

Comment:
Should use Tyk Classic instead of Tyk classic?

Reply:
No, unless we want to change the convention in the future, follows what exists